File: advance/ddd/fizz_buzz.py
def fizz_buzz(limit):
    for i in range(limit):
        if i % 3 == 0:
            print('fizz')
        if i % 5 == 0:
            print('fizz')
        if i % 3 and i % 5:
            print(i)


def main():
    fizz_buzz(10)

#     主函数没有被调用
# 从 0 而不是 1 开始
# 在 15 的整数倍的时候在不用行打印 “fizz” 和 “buzz”
# 在 5 的整数倍的时候打印 “fizz”
# 采用硬编码的参数 10 而不是从命令控制行读取参数

File: languages/java/src/main/java/com/example/tdd/FizzBuzz.java
package com.example.tdd;

/**
 * @author henry
 */
public class FizzBuzz {

    public static void main(String[] args) {
        FizzBuzz1();
    }

    public static void FizzBuzz1() {
        for (int i = 0; i < 200; i++) {
            String result = handleSingleDigit357(i);
            if (result.equals("")) {
                result = String.valueOf(i);
            }

            System.out.println(result);
        }
    }

    public static String handleSingleDigit357(int i) {
        String result = "";
        String s = String.valueOf(i);
        if (i % 3 == 0 || s.contains("3")) {
            result += "Fizz";
        }
        if (i % 5 == 0 || s.contains("5")) {
            result += "Buzz";
        }
        if (i % 7 == 0 || s.contains("7")) {
            result += "Whizz";
        }

        return result;
    }
}
